nuxeo-compound-documents/src/test/resources/operations-test-contrib.xml

```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0"?>
+<component name="org.nuxeo.compound.documents.operations.customisation.test" version="1.0">
+  <require>org.nuxeo.compound.documents.operations</require>
+  <require>org.nuxeo.compound.documents.types</require>
+  <extension target="org.nuxeo.automation.scripting.internals.AutomationScriptingComponent" point="operation">
+    <scriptedOperation id="javascript.GetCompoundDocumentType">
+      <inputType>document</inputType>
+      <outputType>string</outputType>
+      <category>javascript</category>
+      <param name="entries" type="strings" />
+      <script>
+        function run(input, params) {
+          for (var i = 0; i &lt; params.entries.length ; i++)
+          {
+            if (params.entries[i] === "a.txt") {
+              return "CustomCompoundDocument";
+            }
+          }
+          return "CompoundDocument";
+        }
+      </script>
+    </scriptedOperation>
+    <scriptedOperation id="javascript.GetCompoundDocumentFolderType">
+      <inputType>document</inputType>
+      <outputType>string</outputType>
+      <category>javascript</category>
+      <script>
+        function run(input) {
+          if (input.type === "CustomCompoundDocument") {
+            return "CustomCompoundDocumentFolder";
+          }
+          return "CompoundDocumentFolder";
+        }
+      </script>
+    </scriptedOperation>
+  </extension>
+  <extension target="org.nuxeo.ecm.core.schema.TypeService" point="doctype">
+    <doctype name="CustomCompoundDocument" extends="Document">
```

Comment:
Should have the `CompoundDocument` facet to mark it as a compound document.

Reply:
I added a - missing - facet assertion in `org.nuxeo.compound.documents.CompoundDocumentUtils#assertCompoundDocument`